tracer/src/Datadog.Trace.Tools.Runner/Checks/ProcessBasicCheck.cs

```diff
@@ -460,5 +497,169 @@ private static bool ParseBooleanConfigurationValue(string value)
+                {
+                    AnsiConsole.WriteLine(TracingDotnetOnIis);
+                }
+            }
+        }
+
+        private static string? CheckWindowsInstallation()
+        {
+            string? tracerVersion = null;
+            string? displayName = null;
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                const string datadog64BitProgram = "Datadog .NET Tracer 64-bit";
+                const string datadog32BitProgram = "Datadog .NET Tracer 32-bit";
+
+                var registrySubKeys = new[] { @"SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall", @"SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Uninstall" };
+
+                foreach (var registryKey in registrySubKeys)
+                {
+                    using (var key = Registry.LocalMachine.OpenSubKey(registryKey))
+                    {
+                        if (key is not null)
+                        {
+                            foreach (var subKeyName in key.GetSubKeyNames())
+                            {
+                                var subKey = key.OpenSubKey(subKeyName);
+
+                                if (subKey?.GetValue("DisplayName") is datadog64BitProgram or datadog32BitProgram && displayName is null)
+                                {
+                                    displayName = (string?)subKey.GetValue("DisplayName");
+                                    tracerVersion = subKey.GetValue("VersionMajor") + "." + subKey.GetValue("VersionMinor");
+                                    break;
+                                }
+                            }
+                        }
+                    }
+                }
+
+                if (displayName is null)
+                {
+                    Utils.WriteError(TraceProgramNotFound);
+                }
+                else if (displayName is datadog64BitProgram)
+                {
+                    Utils.WriteSuccess(TracerProgramFound(datadog64BitProgram));
+                }
+                else if (displayName is datadog32BitProgram)
+                {
+                    if (RuntimeInformation.OSArchitecture is Architecture.X64)
+                    {
+                        Utils.WriteError(WrongTracerArchitecture(datadog32BitProgram));
+                        return tracerVersion;
+                    }
+
+                    Utils.WriteSuccess(TracerProgramFound(datadog32BitProgram));
+                }
+            }
+
+            return tracerVersion;
+        }
+
+        private static void CheckAppPools(ApplicationPool? pool)
+        {
+            var tracingRelevantVariables = new Dictionary<string, string>
+            {
+                { "COR_ENABLE_PROFILING", "1" },
+                { "CORECLR_ENABLE_PROFILING", "1" },
+                { "COR_PROFILER", Utils.Profilerid },
+                { "CORECLR_PROFILER", Utils.Profilerid }
```

Comment:
Do we need to add the `_32` and `_64` versions? 🤔

Reply:
Hmm what would we do with the path variables if we found them in the environment variables list? I guess we could check that the path points to a file on disk, but we don't have a good way to check that the file it's pointing to is actually our profiler

Reply:
> what would we do with the path variables if we found them in the environment variables list

We can make some guesses though I guess - e.g. we know what our profiler's file name is, and the _default_ install locations + we can check if the file exists. Those all seems useful? 🤔